### ISSUE TITLE
.NET 8: Localize apphost error page

### ISSUE THREAD

**Issue by richlander** [dotnet/runtime#71283]
There was a request on our [host improvements post](https://devblogs.microsoft.com/dotnet/dotnet-apphost-improvements) to [localize our host error messages](https://devblogs.microsoft.com/dotnet/dotnet-apphost-improvements/#comment-15950) (specifically relating to the download the runtime UX). That sounds like a good idea. We should consider it.

@elinor-fung @vitek-karas

**Comment by vitek-karas** [comment]
@elinor-fung did investigation on this topic and wrote some ideas here: https://github.com/dotnet/runtime/blob/main/docs/design/features/localization-options.md

**Comment by deeprobin** [comment]
I have often heard that we want to keep the host relatively small (in size). With the localization option, I think the executable becomes significantly larger.

Is this planned only for the "normal" host or also for the singlefilehost?

**Comment by richlander** [comment]
That thinking makes sense to me.

I am wondering if we want to start with the (shared) dotnet host localization topic first. That would help developers and is still worthwhile.

We might approach the apphost localization topic at the same time as updating the UI. Our approach there might influence the technical localization plan.

I am wondering if we end up with a bare bones experience and then a fancy one and then let people choose. I have this feeling that it is going to be very hard to provide a fancy experience that works for all scenarios.

**Comment by agocke** [comment]
Multiplying hosts is likely to be pretty expensive, partially because once we scale past one or two we'll have to build dedicated infrastructure to handle the building and publishing (not to mention testing).

**Comment by richlander** [comment]
Agreed on that. We need to spend some more time thinking about how to deliver a better experience.
